packages/yourloops/lib/auth/hook.tsx

```diff
@@ -196,24 +187,53 @@ export function AuthContextImpl(): AuthContext {
     }
   }
 
+  const setAccessToken = useCallback((ignoreCache: boolean = false): void => {
+    const getAccessToken = async (): Promise<string> => await getAccessTokenSilently({ ignoreCache })
+    HttpService.setGetAccessTokenMethod(getAccessToken)
+  }, [getAccessTokenSilently])
+
+  /*
+  * This method is a trick to manually refresh auth0 access token.
+  * Auth0 SDK doesn't provide a method to refresh it, it's done automatically when it expires.
+  * But in the case of complete signup we need a new token immediately after updating the role (to get this new role in the token),
+  * so the trick is to call getAccessTokenSilently with ignoreCache option, then make an API call and auth0 will get a fresh new token !!
+   */
+  const refreshAccessToken = async (): Promise<void> => {
```

Comment:
We need to discuss this :slightly_smiling_face:

Reply:
🙃